storage-node/src/services/sync/acceptPendingObjects.ts

```diff
@@ -0,0 +1,212 @@
+import { ApiPromise } from '@polkadot/api'
+import { KeyringPair } from '@polkadot/keyring/types'
+import fs from 'fs'
+import path from 'path'
+import { addDataObjectIdToCache } from '../caching/localDataObjects'
+import { registerNewDataObjectId } from '../caching/newUploads'
+import logger from '../logger'
+import { QueryNodeApi } from '../queryNode/api'
+import { acceptPendingDataObjectsBatch } from '../runtime/extrinsics'
+const fsPromises = fs.promises
+
+export type AcceptPendingDataObjectsParams = {
+  account: KeyringPair
+  storageBucket: {
+    id: string
+    bags: {
+      id: string
+      dataObjects: string[]
+    }[]
+  }
+}
+
+export class AcceptPendingObjectsService {
+  private pendingDataObjects: Map<string, [string, string]> // dataObjectId -> [storageBucketId, bagId]
+
+  private constructor(
+    private qnApi: QueryNodeApi,
+    private pendingDataObjectsDir: string,
+    private bucketKeyPairs: Map<string, KeyringPair>,
+    private uploadBuckets: string[]
+  ) {
+    this.pendingDataObjects = new Map()
+  }
+
+  static async init(
+    api: ApiPromise,
+    qnApi: QueryNodeApi,
+    workerId: number,
+    uploadsDir: string,
+    pendingDataObjectsDir: string,
+    bucketKeyPairs: Map<string, KeyringPair>,
+    uploadBuckets: string[],
+    maxTxBatchSize: number,
+    intervalMs: number
+  ): Promise<AcceptPendingObjectsService> {
+    const acceptPendingObjectsService = new AcceptPendingObjectsService(
+      qnApi,
+      pendingDataObjectsDir,
+      bucketKeyPairs,
+      uploadBuckets
+    )
+
+    // Load pending data objects from the pending directory
+    await acceptPendingObjectsService.loadPendingDataObjects()
+
+    const runWithInterval = () => {
+      acceptPendingObjectsService
+        .acceptPendingDataObjects(api, workerId, uploadsDir, maxTxBatchSize)
+        .catch((err) => logger.error(`Failed to register pending data objects as accepted in runtime: ${err}`))
+
+      setTimeout(runWithInterval, intervalMs)
+    }
+    runWithInterval()
+
+    return acceptPendingObjectsService
+  }
+
+  getPendingDataObject(objectId: string): [string, string] | undefined {
+    return this.pendingDataObjects.get(objectId)
+  }
+
+  private async loadPendingDataObjectsFromIDs(pendingIds: string[]): Promise<void> {
+    const pendingDataObjects = await this.qnApi.getDataObjectsByIds(pendingIds)
+
+    pendingDataObjects.forEach((dataObject) => {
+      const storageBucket = dataObject.storageBag.storageBuckets.find(({ id }) => this.uploadBuckets.includes(id))
+      if (storageBucket) {
+        this.push(dataObject.id, storageBucket.id, dataObject.storageBag.id)
```

Comment:
```suggestion
        this.insert(dataObject.id, storageBucket.id, dataObject.storageBag.id)
```
renaming just for correct symantics given the underlying data structure.